### ISSUE TITLE
TypeError: Unicode-objects must be encoded before hashing in `s3cmd sync --cf-invalidate`

### ISSUE THREAD

**Issue by rkmax** [s3tools/s3cmd#1046]
it's present on the latest version, tested on macos sierra and Arch Linux both on python 3.7
```
Problem: <class 'TypeError: Unicode-objects must be encoded before hashing
S3cmd:   2.0.2
python:   3.7.3 (default, Mar 26 2019, 21:43:19) 
[GCC 8.2.1 20181127]
environment LANG=es_CO.UTF-8

**Comment by sam0x17** [comment]
same issue here on macos mojave 

```
S3cmd:   2.0.2
python:   3.7.3 (default, Jun 19 2019, 07:38:49)
```

**Comment by sam0x17** [comment]
I have confirmed that this remains an issue on any new install of macos using Python 3 or Python 2.7, regardless of whether s3cmd is installed via homebrew or pip. So as of right now s3cmd's cf-invalidate feature is unusable on macos.

**Comment by sam0x17** [comment]
I have confirmed that this happens on freshly installed 2018 macs.

**Comment by sam0x17** [comment]
I've set up https://github.com/sam0x17/s3cmd-shim in the mean time until this get fixed. You can use it as a drop-in replacement for s3cmd on macos (docker-based).

**Comment by squatto** [comment]
This is occurring on Ubuntu on our AWS servers as of today.

`uname -a` = `Linux ip-*-*-*-* 4.15.0-1048-aws #50-Ubuntu SMP Wed Sep 4 22:03:52 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux`

Command run:

`/usr/bin/s3cmd sync --acl-public --delete-removed --follow-symlinks --cf-invalidate /source/folder/ s3://bucket-name/dest/folder/`

Full error:

```
Problem: <class 'TypeError: Unicode-objects must be encoded before hashing
S3cmd:   2.0.1
python:   3.6.8 (default, Aug 20 2019, 17:12:48)
[GCC 8.3.0]
environment LANG=C.UTF-8

Traceback (most recent call last):
  File "/usr/bin/s3cmd", line 3073, in <module>
    rc = main()
  File "/usr/bin/s3cmd", line 2989, in main
    rc = cmd_func(args)
  File "/usr/bin/s3cmd", line 1873, in cmd_sync
    return cmd_sync_local2remote(args)
  File "/usr/bin/s3cmd", line 1852, in cmd_sync_local2remote
    _invalidate_on_cf(destination_base_uri)
  File "/usr/bin/s3cmd", line 1822, in _invalidate_on_cf
    results = cf.InvalidateObjects(destination_base_uri, uploaded_objects_list, default_index_file, cfg.invalidate_default_index_on_cf, cfg.invalidate_default_index_root_on_cf)
  File "/usr/lib/python3/dist-packages/S3/CloudFront.py", line 452, in InvalidateObjects
    cfuris = self.get_dist_name_for_bucket(uri)
  File "/usr/lib/python3/dist-packages/S3/CloudFront.py", line 587, in get_dist_name_for_bucket
    response = self.GetList()
  File "/usr/lib/python3/dist-packages/S3/CloudFront.py", line 323, in GetList
    response = self.send_request("GetList")
  File "/usr/lib/python3/dist-packages/S3/CloudFront.py", line 506, in send_request
    request = self.create_request(operation, dist_id, request_id, headers)
  File "/usr/lib/python3/dist-packages/S3/CloudFront.py", line 555, in create_request
    signature = self.sign_request(headers)
  File "/usr/lib/python3/dist-packages/S3/CloudFront.py", line 567, in sign_request
    signature = sign_string_v2(string_to_sign)
  File "/usr/lib/python3/dist-packages/S3/Crypto.py", line 68, in sign_string_v2
    signature = base64.encodestring(hmac.new(encode_to_s3(secret_key), string_to_sign, sha1).digest()).strip()
  File "/usr/lib/python3.6/hmac.py", line 144, in new
    return HMAC(key, msg, digestmod)
  File "/usr/lib/python3.6/hmac.py", line 84, in __init__
    self.update(msg)
  File "/usr/lib/python3.6/hmac.py", line 93, in update
    self.inner.update(msg)
TypeError: Unicode-objects must be encoded before hashing
```

**Comment by Trane9991** [comment]
@fviard could you please release the new version of `s3cmd` since fix already in master (according to the #1006 ?

**Comment by fviard** [comment]
I close the issue as the fix is already in MASTER.
Release with it to come soon.
